app.py

```
@@ -0,0 +1,229 @@
+            'progress': 0
+        })
+        translation_tasks[task_id].update({
+            'status': 'failed',
+            'error': str(e),
+            'progress': 0
+        })
+
+@app.route('/')
+def index():
+    return render_template('index.html',
+                          models=SUPPORTED_MODELS,
```

Ruff (0.8.2)

Undefined name `SUPPORTED_MODELS`

(F821)